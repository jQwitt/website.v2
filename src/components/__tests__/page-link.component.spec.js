import React from 'react';
import { render, cleanup, fireEvent } from '@testing-library/react';
import { PageLink } from '../index';

afterEach(cleanup);
describe('PageLink', () => {
  test('snapshot', () => {
    const { asFragment } = render(<PageLink />);
    expect(asFragment()).toMatchSnapshot();
  });

  it('renders props correctly', () => {
    const { getByText } = render(<PageLink nameToLink={'test'} />);
    expect(getByText('test')).toBeTruthy();
  });

  it('calls scrollToName', () => {
    const scrollToName = jest.fn();
    const { container, getByText } = render(<PageLink nameToLink={'test'} />);
    fireEvent.click(container.firstChild);
    expect(scrollToName).toHaveBeenCalled;
  });
});
