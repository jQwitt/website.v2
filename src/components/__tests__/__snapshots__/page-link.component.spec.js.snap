// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PageLink snapshot 1`] = `
<DocumentFragment>
  <div>
    <h3 />
  </div>
</DocumentFragment>
`;
